geomesa-gs-monitor-elasticsearch/src/main/scala/org/geomesa/gs/monitor/elastic/ElasticRequestDataListener.scala

```diff
@@ -8,137 +8,165 @@
 
 package org.geomesa.gs.monitor.elastic
 
-import java.lang.reflect.Type
-import java.net.{MalformedURLException, URL}
-import java.util.Date
-
-import com.google.gson._
+import com.typesafe.config.{Config, ConfigFactory}
 import com.typesafe.scalalogging.LazyLogging
 import org.apache.http.HttpHost
-import org.elasticsearch.action.ActionListener
-import org.elasticsearch.client.{RequestOptions, RestClient, RestHighLevelClient}
-import org.elasticsearch.action.index.{IndexRequest, IndexResponse}
-import org.elasticsearch.common.xcontent.XContentType
-import org.elasticsearch.action.DocWriteResponse
-import org.geoserver.monitor.{RequestData, RequestDataListener}
-import org.opengis.geometry.BoundingBox
-
-import scala.collection.mutable.ArrayBuffer
-
-
+import org.apache.http.auth.{AuthScope, UsernamePasswordCredentials}
+import org.apache.http.impl.client.BasicCredentialsProvider
+import org.apache.http.impl.nio.client.HttpAsyncClientBuilder
+import org.elasticsearch.client.RestClientBuilder.HttpClientConfigCallback
+import org.elasticsearch.client.{Request, RestClient}
+import org.geomesa.gs.monitor.elastic.ExtendedRequestData.TIMEOUT_KEY
+import org.geoserver.monitor
+import org.geoserver.monitor.RequestData.Status
+import org.geoserver.monitor.RequestDataListener
+
+import java.util.concurrent.{Executors, LinkedBlockingQueue, TimeUnit}
+import scala.collection.JavaConverters._
+import scala.concurrent.TimeoutException
 
 class ElasticRequestDataListener extends RequestDataListener with LazyLogging {
-  import ElasticRequestDataListener.gson
-  val envVars = sys.env.getOrElse("ELASTICSEARCH_HOST", "unset").split(',')
-  if(envVars(0) == "unset"){
-    logger.error("Environment variable ELASTICSEARCH_HOST is not set")
-    throw new Exception("Environment variable ELASTICSEARCH_HOST is not set. Stopping build.")
-  }
-  //initialized variables
-  var host = ""
-  var port = 0
-  var protocol = ""
-  val hostList = new ArrayBuffer[HttpHost]
-
-  for (u <- envVars) {
-    try {
-      val url = new URL(u.trim())
-      host = url.getHost
-      port = url.getPort
-      protocol = url.getProtocol
-      hostList.append(new HttpHost(host, port, protocol))
-    }
-    catch{
-      case e: MalformedURLException => logger.error("Invalid URL " + u + ". Could not convert from string to URL. Trying any additional URLs")
+
+  import org.geomesa.gs.monitor.elastic.ElasticRequestDataListener._
+
+  private val config = ConfigFactory.load.getConfig(GEOSERVER_MONITOR_ELASTICSEARCH_KEY)
+  private val index = config.getString("index")
+
+  private val excludedFields =
+    Option("excludedFields")
+      .filter(config.hasPath)
+      .map(config.getStringList)
+      .map(_.asScala.toSet)
+      .getOrElse(Set.empty)
+  private val timeout =
+    Option("timeout")
+      .filter(config.hasPath)
+      .map(config.getLong)
+      .filter(ms => ms > 1000L && ms < 1000000L)
```

Comment:
minor - might be good to log something here so the user knows their config is being dropped.